### DISCUSSION TITLE
Can't Trigger button Command

### DISCUSSION THREAD

**Discussion by yuyue5945** [PanAndZoom/PanAndZoom#51] [Q&A]
a button in the panAndZoom,binding a command ， can't trigger the command

out of the panAndZoom zoon, can trigger

how to do to solu the issue

**Comment by wieslawsoltes** [comment]
Can you post a simple repro?

**Reply by yuyue5945** [reply]
Source code

xaml file
<UserControl x:Class="PluginGUI_QinZhou.Yard"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:prism="http://prismlibrary.com/"
        prism:ViewModelLocator.AutoWireViewModel="True"
       xmlns:panAndZoom="clr-namespace:Wpf.Controls.PanAndZoom;assembly=Wpf.Controls.PanAndZoom"

    <Grid>
        <panAndZoom:ZoomBorder Name="ZoomBorder2"  
                               Stretch="None" 
                               ZoomSpeed="1.1"
                               MaxZoomX="2"
                               MaxZoomY="2" 
                               Background="Transparent"
                               EnableConstrains="False"  ClipToBounds="True" Focusable="False"
                               VerticalAlignment="Stretch" HorizontalAlignment="Stretch" >
   
            <Canvas >
                <Button Canvas.Left="100" Canvas.Top="110" Width="150" Command="{Binding AgvDelegateCommandCommand}">
       </Canvas >

    </Grid>

.cs file


    private DelegateCommand agvDelegateCommandCommand;
        public DelegateCommand AgvDelegateCommandCommand
        {
            get { return agvDelegateCommandCommand; }
            set { SetProperty(ref agvDelegateCommandCommand, value); }
        }


    public YardViewModel(IRegionManager regionManager, IEventAggregator eventAggregator) : base(regionManager, eventAggregator)
        {
           
            AgvDelegateCommandCommand = new DelegateCommand(ExcuteAGVCommand, CanExcute_AGVCommand);

        }

   private void ExcuteAGVCommand()
        {
            string a = "123";
        }

        private bool CanExcute_AGVCommand()
        {
            return true;
        }

**Comment by FelixClarkART** [comment]
Any update on this? I'm getting the same thing.

If you take the WPF demo app and add a simple button with a 'Click' event you'll see it doesn't get fired if within the zoom window.

If it's outside it fires as expected. Same thing occurs with an ICommand binding.

**MainWindow.xaml**
`        <paz:ZoomBorder Name="zoomBorder" Stretch="None" ZoomSpeed="1.2" 
                        Background="SlateBlue" ClipToBounds="True" Focusable="True"
                        VerticalAlignment="Stretch" HorizontalAlignment="Stretch" 
                        Grid.Row="4" Grid.Column="1">
            <Canvas Background="LightGray" Width="300" Height="300">
                <Rectangle Canvas.Left="100" Canvas.Top="100" Width="50" Height="50" Fill="Red"/>
                <StackPanel Canvas.Left="100" Canvas.Top="200">
                    <TextBlock Text="Text1" Width="100" Background="Red" Foreground="WhiteSmoke"/>
                    <TextBlock Text="Text2" Width="100" Background="Red" Foreground="WhiteSmoke"/>
                    <Button Click="Button_Click" Height="50" Content="Test" Background="Blue"/>
                </StackPanel>
            </Canvas>
        </paz:ZoomBorder>
        <Button Click="Button_Click" Height="50" Content="Test" Background="Blue"/>`


**MainWindow.xaml.cs**
`            
        private void Button_Click(object sender, RoutedEventArgs e)
        {
            Debug.WriteLine("Button clicked!");
        }`

**Reply by FelixClarkART** [reply]
I can probably just get away with using `PreviewMouseLeftButtonDown` for the button, seems to work.
